### ISSUE TITLE
Remove euclid dependency

### ISSUE THREAD

**Issue by kvark** [servo/surfman#208]
It's unclear what value `euclud` brings here. A cursory look reveals only `Point2D` and `Size2D` used, and `surfman` could just roll out its own structures for this.

**Comment by avr1** [comment]
@kvark I'd like to take a look at this, but I'm a little confused as to how to approach the problem. Should I make my own definitions of the Euclid structures, and if so, where in the codebase should I store them and refer all other calls to those structures to? Thanks!

**Comment by kvark** [comment]
@avr1 Sorry! I stopped using surfman in gfx-rs and opted into talking to EGL directly instead, so I can't help here.

**Comment by avr1** [comment]
Thanks anyway! @jdm, do you think I should make my own definitions of the Euclid structures used, and if so, where do you suggest I put them? If not, should we continue to use euclid?

**Comment by jdm** [comment]
Yeah, I would just put them in the root of surfman, or maybe make a geom module that's publicly visible.

**Comment by avr1** [comment]
@jdm Sorry to ask such a newb question, but where exactly is the root of surfman, and how would I access that in the subdirectories? I've been fighting a losing war with the module system.

**Comment by jdm** [comment]
https://github.com/servo/surfman/blob/master/surfman/src/lib.rs

**Comment by jdm** [comment]
From submodules you could refer to `use crate::{Size2D, Point2D};`
